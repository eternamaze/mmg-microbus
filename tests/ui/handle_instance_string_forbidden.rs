use mmg_microbus::prelude::*;

#[mmg_microbus::component]
struct C { id: mmg_microbus::bus::ComponentId }

#[mmg_microbus::handles]
impl C {
    #[mmg_microbus::handle(Tick, from=Src, instance="x")] // ERROR: string instance forbidden
    async fn on_tick(&mut self, _env: std::sync::Arc<Envelope<Tick>>) -> Result<()> { Ok(()) }
}

#[derive(Clone)]
struct Tick;
struct Src;

fn main() {}
