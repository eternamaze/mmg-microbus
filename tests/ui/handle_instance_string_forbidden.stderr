error: #[handle]: `instance` expects a marker type (impl InstanceMarker)
 --> tests/ui/handle_instance_string_forbidden.rs:8:53
  |
8 |     #[mmg_microbus::handle(Tick, from=Src, instance="x")] // ERROR: string instance forbidden
  |                                                     ^^^
